ThinDownloadManager/src/main/java/com/thin/downloadmanager/DownloadDispatcher.java

```diff
@@ -79,7 +81,7 @@ public void run() {
                 mRedirectionCount = 0;
                 Log.v(TAG, "Download initiated for " + mRequest.getDownloadId());
                 updateDownloadState(DownloadManager.STATUS_STARTED);
-                executeDownload(mRequest.getUri().toString());
+                executeDownload(mRequest.getUri().toString(), mRequest.getCustomHeaders());
```

Comment:
We can the signature of executeDownload method as it is. Since 'mRequest' is available throughout this class, we can use mRequest to get the customHeaders. That is how RetryPolicy is used. For consistency please change this.